packages/gitbook/src/components/Search/SearchContainer.tsx

```diff
@@ -0,0 +1,186 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+import React, { useRef, useState } from 'react';
+import { useHotkeys } from 'react-hotkeys-hook';
+import { useTrackEvent } from '../Insights';
+import { Popover } from '../primitives';
+import { SearchAskAnswer } from './SearchAskAnswer';
+import { useSearchAskState } from './SearchAskContext';
+import { SearchAskProvider } from './SearchAskContext';
+import { SearchInput } from './SearchInput';
+import { SearchResults, type SearchResultsRef } from './SearchResults';
+import { SearchScopeToggle } from './SearchScopeToggle';
+import { useSearch } from './useSearch';
+
+interface SearchContainerProps {
+    spaceTitle: string;
+    isMultiVariants: boolean;
+    withAsk: boolean;
+    withAIChat: boolean;
+    className?: string;
+}
+
+/**
+ * Client component to render the search modal when the url contains a search query.
+ */
+export function SearchContainer(props: SearchContainerProps) {
+    const { spaceTitle, isMultiVariants, withAsk, withAIChat, className } = props;
+
+    const [state, setSearchState] = useSearch();
+    const [open, setOpen] = useState(false);
```

Comment:
Why do we need this state? since we have `useSearch` above which should manage this